Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/Application/AtomToolsApplication.h

```diff
@@ -0,0 +1,111 @@
+        AZ_TYPE_INFO(AtomTools::AtomToolsApplication, "{A0DF25BA-6F74-4F11-9F85-0F99278D5986}");
+
+        using Base = AzFramework::Application;
+
+        AtomToolsApplication(int* argc, char*** argv);
+
+        //////////////////////////////////////////////////////////////////////////
+        // AzFramework::Application
+        void CreateReflectionManager() override;
+        void Reflect(AZ::ReflectContext* context) override;
+        void RegisterCoreComponents() override;
+        AZ::ComponentTypeList GetRequiredSystemComponents() const override;
+        void CreateStaticModules(AZStd::vector<AZ::Module*>& outModules) override;
+        const char* GetCurrentConfigurationName() const override;
+        void StartCommon(AZ::Entity* systemEntity) override;
+        void Tick(float deltaOverride = -1.f) override;
+        void Stop() override;
+
+    protected:
+        //////////////////////////////////////////////////////////////////////////
+        // AssetDatabaseRequestsBus::Handler overrides...
+        bool GetAssetDatabaseLocation(AZStd::string& result) override;
+        //////////////////////////////////////////////////////////////////////////
+
+        //////////////////////////////////////////////////////////////////////////
+        // AzFramework::Application overrides...
+        void Destroy() override;
+        //////////////////////////////////////////////////////////////////////////
+
+        //////////////////////////////////////////////////////////////////////////
+        // AZ::ComponentApplication overrides...
+        void QueryApplicationType(AZ::ApplicationTypeQuery& appType) const override;
+        //////////////////////////////////////////////////////////////////////////
+
+        //////////////////////////////////////////////////////////////////////////
+        // AZ::UserSettingsOwnerRequestBus::Handler overrides...
+        void SaveSettings() override;
+        //////////////////////////////////////////////////////////////////////////
+
+        ////////////////////////////////////////////////////////////////////////
+        // EditorPythonConsoleNotificationBus::Handler overrides...
+        void OnTraceMessage(AZStd::string_view message) override;
+        void OnErrorMessage(AZStd::string_view message) override;
+        void OnExceptionMessage(AZStd::string_view message) override;
+        ////////////////////////////////////////////////////////////////////////
+
+        virtual AZStd::string_view GetBuildTargetName() {return targetName;};
+
+        virtual void LoadSettings();
+        virtual void UnloadSettings();
+        virtual void CompileCriticalAssets(const AZStd::vector<AZStd::string> &assetFiltersArray);
+        virtual void ProcessCommandLine(const AZ::CommandLine& commandLine);
+        virtual bool LaunchDiscoveryService();
+        virtual void StartInternal();
+
+        static void PyIdleWaitFrames(uint32_t frames);
+        void setTargetName(AZStd::string newTargetName);
```

Comment:
According to naming convention, they should start with a capital S 
If you are going to set the target name and use a member variable than you do not need the virtual function.  
But there should be a GetTargetName function as well.  

Except for small or plain old data type variables, you should pass them as a const & to avoid copying data unnecessarily.  

void SetTargetName(const AZStd::string& targetName);

But honestly since you never want somebody to setTargetName more than once you should get rid of this function and pass the targets name into the constructor or just use the virtual GetTargetName function